v1/bank/bank.gunk

```diff
@@ -0,0 +1,355 @@
+//                                 AuthorizationURL: "/v1/auth",
+//                                 TokenURL:         "/v1/token",
+//                                 Scopes: openapiv2.Scopes{
+//                                         Scope: map[string]string{
+//                                                 "read":  "Grants read access",
+//                                                 "write": "Grants write access",
+//                                         },
+//                                 },
+//                         },
+//                 },
+//         },
+// }
+package bank
+
+import (
+	"github.com/gunk/opt/http"
+	"github.com/gunk/opt/openapiv2"
+
+	types "github.com/openbank/openbank/v1/types"
+)
+
+// BankRouting holds all the details about a bank routing.
+type BankRouting struct {
+	// Scheme is the routing scheme.
+	Scheme string `pb:"1" json:"scheme"`
+
+	// Address is the routing address.
+	Address string `pb:"2" json:"address"`
+}
+
+// Bank holds all details about a bank.
+type Bank struct {
+	// ID is an identifier for the bank.
+	ID string `pb:"1" json:"id"`
+
+	// FullName is the full name of the bank.
+	FullName string `pb:"2" json:"full_name"`
+
+	// ShortName is the short name of the bank.
+	ShortName string `pb:"3" json:"short_name"`
+
+	// LogoURL is the url for the bank's logo.
+	LogoURL string `pb:"4" json:"logo_url"`
+
+	// WebsiteURL is the url for the bank's website.
+	WebsiteURL string `pb:"5" json:"website_url"`
+
+	// SwiftBIC is the SWIFT bank identifier code.
+	SwiftBIC string `pb:"6" json:"swift_bic"`
+
+	// NationalIdentifier is the national identifier code.
+	NationalIdentifier string `pb:"7" json:"national_identifier"`
+
+	// BankRouting is the routing information for the bank.
+	BankRouting BankRouting `pb:"8" json:"bank_routing"`
+}
+
+// TransactionType is holds the information about transaction type at a bank.
+type TransactionType struct {
+}
+
+// CreateBankRequest is a request envelope to create a bank.
+type CreateBankRequest struct {
+	// Bank is the related information about a bank.
+	Bank Bank `pb:"1" json:"bank"`
+}
+
+// UpdateBankRequest is the request envelope to update a bank.
+type UpdateBankRequest struct {
+	// Bank is the related information about a bank.
+	Bank Bank `pb:"1" json:"bank"`
+}
+
+// DeleteBankRequest is the request envelope to delete a bank.
+type DeleteBankRequest struct {
+	// ID is the bank unique identifier.
+	ID string `pb:"1" json:"id"`
+}
+
+// GetBankRequest is the request envelope to get the bank data.
+type GetBankRequest struct {
+	// ID is the bank unique identifier.
+	ID string `pb:"1" json:"id"`
+
+	// FullName is the full name of the bank.
+	FullName string `pb:"2" json:"full_name"`
+
+	// ShortName is the short name of the bank.
+	ShortName string `pb:"3" json:"short_name"`
+
+	// LogoURL is the url for the bank's logo.
+	LogoURL string `pb:"4" json:"logo_url"`
```

Comment:
Do we really need to search for bank by logo URL? :P

Reply:
Good catch, I was copying from the payload based on https://apiexplorer.openbankproject.com/?version=OBPv4.0.0&list-all-banks=false&core=&psd2=&obwg=&ignoredefcat=&tags=&bank_id=aachener-bank-aachen#OBPv3_0_0-bankById, which specify logoURL as a search parameter but it really does not make sense to have this. Thanks for the review guys.